packages/report-server/src/reportBuilder/fetch/functions/dataGroup.ts

```diff
@@ -29,26 +29,32 @@ const fetchEvents = async (
   query: FetchReportQuery,
   params: DataGroupFetchParams,
 ): Promise<FetchResponse> => {
-  const { dataGroupCode, dataElementCodes, aggregations } = params;
+  const { dataGroupCodes, dataElementCodes, aggregations } = params;
   const { organisationUnitCodes, hierarchy, period, startDate, endDate } = query;
-  const response = await aggregator.fetchEvents(
-    dataGroupCode,
-    aggregations,
-    organisationUnitCodes,
-    hierarchy,
-    {
-      period,
-      startDate,
-      endDate,
-    },
-    dataElementCodes,
+  // TODO: Eventually we want Aggregator to handle fetching multiple dataGroups
+  const results = await Promise.all(
+    dataGroupCodes.map(async dataGroupCode => {
+      const response = await aggregator.fetchEvents(
+        dataGroupCode,
+        aggregations,
+        organisationUnitCodes,
+        hierarchy,
+        {
+          period,
+          startDate,
+          endDate,
+        },
+        dataElementCodes,
+      );
+      return response.map(event => {
+        const { dataValues, ...restOfEvent } = event;
+        return { ...dataValues, ...restOfEvent };
+      });
+    }),
   );
-  const rows = response.map(event => {
-    const { dataValues, ...restOfEvent } = event;
-    return { ...dataValues, ...restOfEvent };
-  });
+
   return {
-    results: rows,
+    results: results.flat(),
```

Comment:
I tried to use .flatMap() but it seems like TypeScript having trouble mapping the types here and throwing a type error.